parser/src/java/org/apache/hadoop/hive/ql/parse/IdentifiersParser.g

```diff
@@ -734,6 +734,21 @@ dropPartitionOperator
     EQUAL | NOTEQUAL | LESSTHANOREQUALTO | LESSTHAN | GREATERTHANOREQUALTO | GREATERTHAN
     ;
 
+filterPartitionSpec
+    :
+    LPAREN filterPartitionVal (COMMA  filterPartitionVal )* RPAREN -> ^(TOK_PARTSPEC filterPartitionVal +)
+    ;
+
+filterPartitionVal
+    :
+    identifier filterPartitionOperator constant -> ^(TOK_PARTVAL identifier filterPartitionOperator constant)
+    ;
+
+filterPartitionOperator
+    :
+    EQUAL | NOTEQUAL | LESSTHANOREQUALTO | LESSTHAN | GREATERTHANOREQUALTO | GREATERTHAN | KW_LIKE
```

Comment:
`dropPartitionSpec` seems to use almost the same construct ; I don't see any reason to duplicate it ...
the only difference I see right now is `LIKE` - are there any other differences?

I think instead of duplicate we should use the same stuff...

Reply:
There is one slight difference apart from the LIKE operator, dropPartitionSpec expects PARTITON keyword, But since in MSCK we already have a PARTITONS keyword in the syntax it will cause a problem.

Reply:
I may just blur the distinction between PARTITION/PARTITIONS :)

Reply:
Done!